containers/unit_tests/TestVector.hpp

```diff
@@ -52,6 +52,125 @@ namespace Test {
 
 namespace Impl {
 
+template <typename Scalar, class Device>
+struct test_vector_insert {
+  typedef Scalar scalar_type;
+  typedef Device execution_space;
+
+  template <typename Vector>
+  void run_test(Vector& a) {
+    int n = a.size();
+
+    auto it = a.begin();
+    it += 15;
+    ASSERT_EQ(*it, scalar_type(1));
+
+    auto it_return = a.insert(it, scalar_type(3));
+    ASSERT_EQ(a.size(), n + 1);
+    ASSERT_EQ(std::distance(it_return, a.begin() + 15), 0);
```

Comment:
```suggestion
    ASSERT_EQ(it_return, a.begin() + 15);
```

Reply:
why? shouldn't the iterators be the same?